### PR TITLE
new role for dumping engine and dwh databases

### PR DESCRIPTION
**Pull request by leistnerova**
ansible role that gets database variables (dabatase name, user, ...) and dump files for engine and DWH

### CODE REVIEW

File: roles/ovirt-engine-db-dump/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+ovirt_engine_etc_conf_path: "/etc/ovirt-engine/engine.conf.d"
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/defaults/main.yml]: 's;etc/ovirt-engine/engine.conf.d;etc/ovirt-engine/engine.conf.d/;'

File: roles/ovirt-engine-db-dump/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+---
+# get engine and dwh database settings
+- name: set variables
+  set_fact:
+    ovirt_engine_db_dump_engine_db: {}
+    ovirt_engine_db_dump_dwh_db: {}
+
+- name: get engine variables
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: grep engine variables

File: roles/ovirt-engine-db-dump/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  shell: grep "^ENGINE_DB" {{ ovirt_engine_etc_conf_path }}/10-setup-database.conf
+  register: cmd
+
+- name: get engine variables 2
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: set facts for engine variables

File: roles/ovirt-engine-db-dump/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    ovirt_engine_db_dump_engine_db: >
+      {{
+        ovirt_engine_db_dump_engine_db |
+        combine(
+          dict([ item.partition('=')[::2] ])
+        )
+      }}
+  with_items: "{{ cmd.stdout_lines|list }}"
+
+- name: get dwh variables
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: s/get/grep/

File: roles/ovirt-engine-db-dump/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  shell: grep "^DWH_DB" {{ ovirt_engine_etc_conf_path }}/10-setup-dwh-database.conf
+  register: cmd
+
+- name: get dwh variables 2
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: set facts for dwh variables

File: roles/ovirt-engine-db-dump/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        ovirt_engine_db_dump_engine_db |
+        combine(
+          dict([ item.partition('=')[::2] ])
+        )
+      }}
+  with_items: "{{ cmd.stdout_lines|list }}"
+
+- name: get dwh variables
+  shell: grep "^DWH_DB" {{ ovirt_engine_etc_conf_path }}/10-setup-dwh-database.conf
+  register: cmd
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: should be done only when DWH is enabled?

File: roles/ovirt-engine-db-dump/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+- name: get dwh variables 2
+  set_fact:
+    ovirt_engine_db_dump_dwh_db: >
+      {{
+        ovirt_engine_db_dump_dwh_db |
+        combine(
+          dict([ item.partition('=')[::2] ])
+        )
+      }}
+  with_items: "{{ cmd.stdout_lines|list }}"
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: should be done only when DWH is enabled?

File: roles/ovirt-engine-db-dump/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+- name: stop engine service
+  service:
+    name: ovirt-engine
+    state: stopped
+
+- name: stop dwh service
+  service:
+    name: ovirt-engine-dwhd
+    state: stopped
+  when: ovirt_engine_db_dump_dwh == True
```
Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: I believe we should stop dwh service even if we are not dumping dwh database, and do it before stopping engine service

Comment (leistnerova) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: I'm not sure about that, there will be a gap in samples history I assume. And I think there is no reason for that when we are not using the dwh database.
Do it before engine service is good notice.

Comment (StLuke) [inline roles/ovirt-engine-db-dump/tasks/main.yml]: should be stopped with engine anyway, discussed offline

**Comment by leistnerova** [comment]
This role will be part of new role ovirt-engine-local-to-remote-db that I'm working on now and that I'll add to tests. Will that be enough?

**Comment by lukas-bednar** [comment]
> This role will be part of new role ovirt-engine-local-to-remote-db that I'm working on now and that I'll add to tests. Will that be enough?

ok then ... just make sure that at some point this role is included in tests that we can see issues in future PRs

**Pull request merged**